src/worker/startup.ts

```diff
@@ -154,15 +163,23 @@ export async function stop() {
 	// Stop sending metrics for queues
 	clearInterval(timer);
 
-	// Close all queues
+	// Pauses message processing in all queues globally.
+	// The first parameter of this function "isLocal" controls if all queues will be paused
+	await queues.discovery.pause(false, true)
+
+	sqsQueues.stop();
+
+	running = false;
+}
+
+/**
+ * Shuts down the worker and releases all resources. Currently used to cleanup after tests
+ */
+export async function shutdown() {
 	await Promise.all([
 		queues.discovery.close(),
 		queues.installation.close(),
 		queues.push.close(),
 		queues.metrics.close()
 	]);
```

Comment:
If we're just pausing, do we need shutdown?  There's no way of restarting the redis connection without recreating the queues from scratch.

Reply:
As the function comment says, we only use it for tests cleanup. We don't re-create queues in production